git_machete/client.py

```diff
@@ -1681,16 +1681,24 @@ def checkout_github_prs(self, pr_no: int) -> None:
         repo: str
         remote: str
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        current_user: Optional[str] = git_machete.github.derive_current_user_login()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
         print(f"Fetching {remote}...")
         self.__git.fetch_remote(remote)
-        self.flush_caches()
 
         pr = get_pull_request_by_number_or_none(pr_no, org, repo)
         if not pr:
             raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
         if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
-            raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote}`.")
+            remote_from_pr: str = pr.repo['full_name'].split('/')[0]
+            try:
+                self.__git.add_remote(remote_from_pr, pr.repo['html_url'])
+            except MacheteException:  # TODO (#164): make dedicated exception here
```

Comment:
A matter of code style (rather than soundness), but I'd avoid using an exception in this way (as a way to capture a situation which could be easily prevented in advance with a simple `if`)... I'd rather check for the remote's existence before trying to add